installer/ovatools/vic-ova-ui/main.go

```diff
@@ -0,0 +1,222 @@
+	// Fetch OVF Environment via RPC
+	ovfEnv, err := config.String("guestinfo.ovfEnv", "")
+	if err != nil {
+		fmt.Println("impossible to fetch ovf environment, exiting")
+		os.Exit(1)
+	}
+
+	// TODO: fix this when proper support for namespaces is added to golang.
+	// ref: golang/go/issues/14407 and golang/go/issues/14407
+	ovfEnv = strings.Replace(ovfEnv, "oe:key", "key", -1)
+	ovfEnv = strings.Replace(ovfEnv, "oe:value", "value", -1)
+
+	var ovf environment
+
+	var ips string
+
+	err = xml.Unmarshal([]byte(ovfEnv), &ovf)
+	if err != nil {
+		ips = fmt.Sprintf("error: %s\n", err.Error())
+	}
+
+	ifaces, _ := net.Interfaces()
+	for _, i := range ifaces {
+		addrs, _ := i.Addrs()
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+			// If the IP is a loopback interface, coming from docker0 or an ip6, we don't need it
+			if ip.IsLoopback() || ip.String() == "172.17.0.1" || ip.To4() == nil {
+				continue
+			}
+			if strings.ToLower(ovf.Properties["harbor.deploy"]) == "true" {
+				ips = fmt.Sprintf("%sAccess the Container Registry at:\nhttps://%s:%s\n", ips, ip.String(), ovf.Properties["harbor.port"])
+			}
+			if strings.ToLower(ovf.Properties["admiral.deploy"]) == "true" {
+				ips = fmt.Sprintf("%sAccess the Container Management Portal at:\nhttps://%s:%s\n", ips, ip.String(), ovf.Properties["admiral.port"])
+			}
+		}
+	}
+
+	bottompanel := ui.NewPar(ips)
+	bottompanel.Height = ui.TermHeight()/2 + 1
+	bottompanel.Width = ui.TermWidth()
+	bottompanel.TextFgColor = ui.ColorBlack
+	bottompanel.TextBgColor = blue
+	bottompanel.Y = ui.TermHeight() / 2
+	bottompanel.X = 0
+	bottompanel.Bg = blue
+	bottompanel.BorderFg = ui.ColorWhite
+	bottompanel.BorderBg = blue
+	bottompanel.BorderTop = false
+	bottompanel.PaddingTop = 1
+	bottompanel.PaddingLeft = 4
+
+	toppanel := ui.NewPar(fmt.Sprintf("VMware vSphere Integrated Containers %s\n\n%s\n%s", version.GetBuild().ShortVersion(), getCPUs(), getMemory()))
+	toppanel.Height = ui.TermHeight()/2 + 1
+	toppanel.Width = ui.TermWidth()
+	toppanel.TextFgColor = ui.ColorWhite
+	toppanel.Y = 0
+	toppanel.X = 0
+	toppanel.TextBgColor = gray
+	toppanel.Bg = gray
+	toppanel.BorderBg = gray
+	toppanel.BorderFg = ui.ColorWhite
+	toppanel.BorderBottom = false
+	toppanel.PaddingTop = 4
+	toppanel.PaddingLeft = 4
+
+	ui.Handle("/sys/kbd/q", func(ui.Event) {
+		ui.StopLoop()
+	})
+
+	ui.Render(toppanel, bottompanel)
+
+	ui.Loop()
+}
+
+type environment struct {
+	Properties map[string]string
+}
+
+func (e *environment) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+
+	type property struct {
+		Key   string `xml:"key,attr"`
+		Value string `xml:"value,attr"`
+	}
+
+	type propertySection struct {
+		Property []property `xml:"Property"`
+	}
+
+	var environment struct {
+		XMLName         xml.Name        `xml:"Environment"`
+		PropertySection propertySection `xml:"PropertySection"`
```

Comment:
There is a an ovf/env package in govmomi, maybe we could use/extend that? https://github.com/vmware/govmomi/blob/master/ovf/env.go#L51